File: src/main/java/org/example/kalkulationsprogramm/domain/AnfrageGeschaeftsdokument.java
package org.example.kalkulationsprogramm.domain;

import java.math.BigDecimal;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "anfrage_geschaeftsdokument")
@Getter
@Setter
public class AnfrageGeschaeftsdokument extends AnfrageDokument {
    @Column(nullable = false)
    private String dokumentid;
    @Column(nullable = false)
    private String geschaeftsdokumentart;
    private BigDecimal bruttoBetrag;
}

File: src/main/kotlin/org/example/kalkulationsprogramm/domain/AnfrageGeschaeftsdokument.kt
package org.example.kalkulationsprogramm.domain

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Table
import java.math.BigDecimal

@Entity
@Table(name = "anfrage_geschaeftsdokument")
open class AnfrageGeschaeftsdokument : AnfrageDokument() {
    @Column(nullable = false)
    open var dokumentid: String? = null

    @Column(nullable = false)
    open var geschaeftsdokumentart: String? = null

    open var bruttoBetrag: BigDecimal? = null

}